app/src/main/java/de/westnordost/streetcomplete/quests/address/AddHousenumberForm.kt

```diff
@@ -41,6 +42,7 @@ class AddHousenumberForm : AbstractQuestFormAnswerFragment<HousenumberAnswer>()
     private var subtractButton: View? = null
 
     private var isHousename: Boolean = false
+    private var isHousenameAndHousenumber: Boolean = false
```

Comment:
Is it possible to use a single enum variable?

Now there are four possible options (false+false, false+true, true+false, true+true), while only three are used (false+false, false+true, true+false).

An enum variable would make that more clear and also allow Kotlin / the IDE to suggest exhaustive `when` branches.

Reply:
Good idea, though `savedInstanceState` will be trickier (I need to find out how to save that - probably as a string?)

Reply:
String or Int. Both have:
```kt
enum class HouseAddressType { NUMBER, NAME, NAME_AND_NUMBER }
private var addressType: HouseAddressType = HouseAddressType.NUMBER
```
Then String looks something like:
```kt
// serialize
outState.putString(ADDRESS_TYPE, addressType.name)
// deserialize
addressType = try {
    HouseAddressType.valueOf(savedInstanceState?.getString(ADDRESS_TYPE))
} catch (e: IllegalArgumentException) {
    HouseAddressType.NUMBER
}
```
And Int looks something like:
```kt
// serialize
outState.putString(ADDRESS_TYPE, addressType.name)
// deserialize
val ordinal = savedInstanceState?.getInt(ADDRESS_TYPE) ?: HouseAddressType.NUMBER.ordinal
addressType = HouseAddressType.values().getOrElse(ordinal, HouseAddressType.NUMBER)
```

ref: https://kotlinlang.org/docs/enum-classes.html#working-with-enum-constants

Reply:
Not entirely happy about savedInstanceState - maybe can be improved further - but done.

Reply:
serializing enums according to their ordinal is general bad practice. In this case, it doesn't matter, but one should not get into the habit of doing it. 

I added a new comment regarding this, resolving this one.